### PR TITLE
WAIV - [2713, 2863, 2913]

### PR DESCRIPTION
**Pull request by LucyHrinishyna**
Fixes #

### Changes

Summary of changes you made.

* Change 1
* Change 2

### Test plan

Explain how to test changes you made.

* [ ] Step 1
* [ ] Step 2

### Demo (optional)

Sreenshots (before & after) or video that show result of your work.

### CODE REVIEW

File: src/client/objects/WaivioObject.js [hunk 1]
```diff
@@ -15,7 +15,9 @@ import './WaivioObject.less';
 const WaivioObject = ({ wobj, unfollow, follow }) => {
   const { usedLocale } = useContext(AppSharedContext);
   const wObject = getClientWObj(wobj, usedLocale);
```
Comment (LucyHrinishyna) [inline src/client/objects/WaivioObject.js:17]: посмотреть

File: src/client/app/Sidebar/ObjectInfo.js [hunk 2]
```diff
@@ -203,26 +197,11 @@ class ObjectInfo extends React.Component {
...
+    const parent = get(this.props.wobject, 'parent');
     return (
-      this.state.parentWobj && (
-        <ObjectCard
-          key={this.state.parentWobj.author_permlink}
-          wobject={this.state.parentWobj}
-          showFollow={false}
-        />
+      parent && (
+        <ObjectCard key={parent.author_permlink} wobject={parent.parentWobj} showFollow={false} />
```
Comment (LucyHrinishyna) [inline src/client/app/Sidebar/ObjectInfo.js:204]: parent

File: src/client/components/Navigation/Topnav.js [hunk 3]
```diff
@@ -630,11 +627,7 @@ class Topnav extends React.Component {
                 <ObjectAvatar item={option} size={40} />
                 <div>
                   <div className="Topnav__search-content">{wobjName}</div>
-                  {parent && (
-                    <div className="Topnav__search-content-small">
-                      {getFieldWithMaxWeight(parent, objectFields.name)}
-                    </div>
-                  )}
+                  {parent && <div className="Topnav__search-content-small">{parent.name}</div>}
```
Comment (LucyHrinishyna) [inline src/client/components/Navigation/Topnav.js:630]: default_name

File: src/client/components/ObjectAvatar.js [hunk 4]
```diff
@@ -46,7 +37,10 @@ const ObjectAvatar = ({ item, size }) => {
 };
 
 ObjectAvatar.propTypes = {
-  item: PropTypes.shape({ parent: PropTypes.oneOfType([PropTypes.shape(), PropTypes.string]) }),
+  item: PropTypes.shape({
+    parent: PropTypes.oneOfType([PropTypes.shape(), PropTypes.string]),
```
Comment (milklim) [inline src/client/components/ObjectAvatar.js:41]: check parent type: .oneOfType() or just .shape()

File: src/client/components/Sidebar/ObjectCard.js [hunk 5]
```diff
@@ -3,45 +3,48 @@ import PropTypes from 'prop-types';
...
 import ObjectAvatar from '../ObjectAvatar';
 import FollowButton from '../../widgets/FollowButton';
-import { getApprovedField } from '../../helpers/wObjectHelper';
 
 import './ObjectCard.less';
 
 const ObjectCard = ({ wobject, alt, showFollow, isNewWindow, unfollow, follow }) => {
-  const name = getApprovedField(wobject, 'name') || wobject.default_name;
-  const pathname = `/object/${wobject.author_permlink}`;
+  if (wobject) {
```
Comment (milklim) [inline src/client/components/Sidebar/ObjectCard.js:10]: use _.isEmpty() to check if wobject exist

File: src/client/object/WobjHistory.js [hunk 6]
```diff
@@ -170,19 +145,13 @@ class WobjHistory extends React.Component {
...
         default:
-          return wobj.fields.sort(
-            (before, after) => Date.parse(after.created) - Date.parse(before.created),
-          );
+          return wobj.fields.sort((before, after) => after.createdAt - before.createdAt);
       }
     };
     let content = object && object.fields && sortedList(object);
-    const isFetching = content && content.length && content[0].append_field_name;
+    const isFetching = content && content.length && content[0].name;
```
Comment (milklim) [inline src/client/object/WobjHistory.js:154]: rename to 'isFetched'

File: src/client/object/WobjHistory.js [hunk 7]
```diff
@@ -192,31 +161,26 @@ class WobjHistory extends React.Component {
       return true;
     });
 
-    if (params[1] && content && content.length && content[0].append_field_name) {
-      content = content.filter(
-        f =>
-          sortingMenuName[params[1]] === f.append_field_name || f.append_field_name === params[1],
-      );
+    if (params[1] && content && content.length && content[0].name) {
```
Comment (milklim) [inline src/client/object/WobjHistory.js:164]: if (params[1] && isFetched)

File: src/client/objectCard/ObjectCardView.js [hunk 8]
```diff
@@ -83,12 +85,12 @@ const ObjectCardView = ({
...
           <div className="ObjectCardView__content-row">
             <Link to={pathName} title={goToObjTitle(objName)} className="ObjectCardView__avatar">
-              {avatarLayout(wObject.avatar)}
+              {avatarLayout()}
             </Link>
             <div className="ObjectCardView__info">
               {parentName && (
                 <Link
-                  to={`/object/${!isEmpty(wObject.parent) ? wObject.parent.author_permlink : null}`}
+                  to={`/object/${!isEmpty(parent) && parent.author_permlink}`}
```
Comment (milklim) [inline src/client/objectCard/ObjectCardView.js:93]: use _.get()

File: src/client/objects/WaivioObject.js [hunk 9]
```diff
@@ -7,15 +7,13 @@ import WeightTag from '../components/WeightTag';
...
 
 import './WaivioObject.less';
 
 const WaivioObject = ({ wobj, unfollow, follow }) => {
-  const { usedLocale } = useContext(AppSharedContext);
-  const wObject = getClientWObj(wobj, usedLocale);
-  const { address, default_name: defaultName, name, website } = wObject;
+  const { default_name: defaultName, name } = wobj;
+  const address = wobj.address && JSON.parse(wobj.address);
+  const website = wobj.website && JSON.parse(wobj.website);
```
Comment (milklim) [inline src/client/objects/WaivioObject.js:16]: Add error handling - JSON.parse() is not safe operation
You can create helper function parseWobjectField(wobject, fieldName)

File: src/client/user/UserWallet.js [hunk 10]
```diff
@@ -91,7 +95,7 @@ class Wallet extends Component {
     isCurrentUser: PropTypes.bool,
     authenticatedUserName: PropTypes.string,
     screenSize: PropTypes.string.isRequired,
-    transactionsHistory: PropTypes.arrayOf(PropTypes.shape()),
+    transactionsHistory: PropTypes.oneOf([PropTypes.arrayOf(PropTypes.shape()), PropTypes.shape()]),
```
Comment (milklim) [inline src/client/user/UserWallet.js:98]: oneOfType

**Pull request merged**